ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -152,6 +160,49 @@ interface ProfilePageState {
   updating: boolean;
 }
 
+interface CompletionTime {
+  completionTime: number;
+  bypassTime: number;
+}
+
+const getRegistrationStatus = (completionTime: number, bypassTime: number) => {
+  return completionTime !== null && completionTime !== undefined ? RegistrationStepStatus.COMPLETED :
+  bypassTime !== null && completionTime !== undefined ? RegistrationStepStatus.BYPASSED : RegistrationStepStatus.UNCOMPLETE;
+};
+
+const bypassedText = (bypassTime: number): JSX.Element => {
+  return <React.Fragment>
+  <div>Bypassed on:</div>
+  <div>{displayDateWithoutHours(bypassTime)}</div>
+</React.Fragment>;
+};
+
+const getCompleteOrBypassContent = ({bypassTime, completionTime}: CompletionTime): JSX.Element => {
+  switch (getRegistrationStatus(completionTime, bypassTime)) {
+    case RegistrationStepStatus.COMPLETED:
+      return <React.Fragment>
+      <div>Completed on:</div>
+      <div>{displayDateWithoutHours(completionTime)}</div>
+    </React.Fragment>;
+    case RegistrationStepStatus.BYPASSED:
+      return bypassedText(bypassTime);
+    default:
+      return;
+  }
+};
+
+const focusCompletionProps = lensOnProps(['completionTime', 'bypassTime']);
+
+const getTwoFactorContent = fp.flow(
```

Comment:
These new functions are very nice, thanks.  Would it be possible to use this approach for all 5 cards?

Reply:
Yes, it is a step in that direction. The language varies a bit between cards. If we standardized on the verbiage, or used an L10N type setup with this text we could probably remove a lot of code.